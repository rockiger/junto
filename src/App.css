:root {
  --bg-color: #f1f3f4;
  --border-color: #dadce0;
  --border-radius: 8px;
  --hover-bg-color: #e8f0fe;
  --link-color: #3c4043;

}

.App-main {
  margin-top: 64px;
  display: flex;
}
.App-sidebar {
  min-width: 258px;
  padding: 1.5rem 0 1.5rem 1rem;
}
.App-main-content {
  flex-grow: 1;
}

input:focus {
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.3) !important;
  border: 1px solid #4d90fe !important;
  outline: none !important;
}