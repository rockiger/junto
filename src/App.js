// @ts-nocheck
import React from 'reactn'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import CssBaseline from '@material-ui/core/CssBaseline'
import { ThemeProvider } from '@material-ui/styles'

import { CreateNewWiki } from 'components/CreateNewWiki'
import { Drive } from './components/Drive/index'
import Navbar from './components/Navbar'
import GoogleLogin from './components/googleLogin'
import Sidebar from './components/Sidebar'
import Home from './components/Home'
import Page from './components/Page'
import { SharedWithMe } from 'components/SharedWithMe'
import { initGA, setGA } from './components/Tracking'

import {
    API_KEY,
    CLIENT_ID,
    DISCOVERY_DOCS,
    SCOPES,
    THEME,
} from './lib/constants'

import './App.css'

class App extends React.Component {
    componentDidMount() {
        initGA('UA-151325933-1')
        setGA({ anonymizeIp: true })
    }

    setGoToNewFile = goToNewFile => this.setState({ goToNewFile })

    setIsSigningIn = isSigningIn => this.setState({ isSigningIn })

    onLogout = response => {
        this.setState({ isSignedIn: false })
    }

    render() {
        return (
            <ThemeProvider theme={THEME}>
                <Router>
                    <CssBaseline />
                    <div className="App">
                        <header className="App-header">
                            <Navbar isSignedIn={this.global.isSignedIn}>
                                <GoogleLogin
                                    clientId={CLIENT_ID}
                                    apiKey={API_KEY}
                                    discoveryDocs={DISCOVERY_DOCS}
                                    scope={SCOPES}
                                    buttonText="Login"
                                />
                            </Navbar>
                        </header>
                        <main className="App-main">
                            {this.global.isSignedIn && (
                                <aside className="App-sidebar">
                                    <Sidebar
                                        goToNewFile={this.global.goToNewFile}
                                        setGoToNewFile={this.setGoToNewFile}
                                    />
                                </aside>
                            )}
                            <div className="App-main-content">
                                <Route
                                    exact
                                    path="/"
                                    render={props => (
                                        <Home
                                            {...props}
                                            isCreatingNewFile={
                                                this.global.isCreatingNewFile
                                            }
                                            isSignedIn={this.global.isSignedIn}
                                            isSigningIn={
                                                this.global.isSigningIn
                                            }
                                            setGoToNewFile={this.setGoToNewFile}
                                        />
                                    )}
                                />
                                <Route
                                    exact
                                    path="/page/:id"
                                    render={props => (
                                        <Page
                                            {...props}
                                            isCreatingNewFile={
                                                this.global.isCreatingNewFile
                                            }
                                            isSignedIn={this.global.isSignedIn}
                                            isSigningIn={
                                                this.global.isSigningIn
                                            }
                                            setGoToNewFile={this.setGoToNewFile}
                                        />
                                    )}
                                />
                                <Route
                                    exact
                                    path="/shared-with-me"
                                    render={props => (
                                        <SharedWithMe
                                            {...props}
                                            isCreatingNewFile={
                                                this.global.isCreatingNewFile
                                            }
                                            isSignedIn={this.global.isSignedIn}
                                            isSigningIn={
                                                this.global.isSigningIn
                                            }
                                            setGoToNewFile={this.setGoToNewFile}
                                        />
                                    )}
                                />
                                <Route
                                    exact
                                    path="/drive/"
                                    render={props => <Drive />}
                                />
                                <Route
                                    exact
                                    path="/new/"
                                    render={props => (
                                        <CreateNewWiki
                                            isSignedIn={this.global.isSignedIn}
                                            isSigningIn={
                                                this.global.isSigningIn
                                            }
                                        />
                                    )}
                                />
                            </div>
                        </main>
                    </div>
                </Router>
            </ThemeProvider>
        )
    }
}

export default App
