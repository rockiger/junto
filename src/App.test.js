import React from 'react'
import ReactDOM from 'react-dom'
import { render, fireEvent, cleanup } from '@testing-library/react'

import App from './App'

afterEach(cleanup)

it('Spinner shows after button clickte', () => {
    const { getByText, debug, container } = render(<App />)
    expect(getByText(/Knowledge/i).textContent).toBe(
        'Knowledge Base for Google Drive!'
    )
    fireEvent.click(getByText('Login with Google'))
    debug()
    //expect(getByText(/Knowledge/i).textContent).toBe(null)
})
