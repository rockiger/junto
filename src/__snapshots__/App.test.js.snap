// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <nav
      className="Nav"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <span
          className="Nav-logo"
        >
          <img
            alt="Logo"
            src="logo.svg"
          />
          <span
            className="Nav-logo-name"
          >
            Awiki
          </span>
        </span>
      </a>
      <div
        className="Nav-search"
      />
      <div
        className="Nav-user"
      >
        <button
          className="action"
          id="authorize_button"
          onClick={[Function]}
        >
          Login
        </button>
      </div>
      <style>
        
            :global(body) {
                margin: 0;
                font-family: 'Open Sans', -apple-system, BlinkMacSystemFont,
                    Avenir Next, Avenir, Helvetica, sans-serif;
            }
            :global(main) {
                margin-top: 64px;
            }
            .Nav {
                position: fixed;
                top: 0;
                display: flex;
                padding: 8px 16px;
                font: 13px/27px Roboto, RobotoDraft, Arial, sans-serif;
                font-family: 'Google Sans', Roboto, Arial, sans-serif;
                flex-grow: 1;
                flex-shrink: 0;
                flex-basis: auto;
                border-bottom: 1px solid #dadce0;
                width: 100%;
                box-sizing: border-box;
                background-color: white;
                z-index: 1000;
            }
            .Nav-logo {
                height: 48px;
                min-width: 238px;
                flex-grow: 1;
                flex-shrink: 0;
                flex-basis: auto;
                display: flex;
                align-items: center;
            }
            .Nav-logo-name {
                display: inline-block;
                font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI',
                    Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans',
                    'Helvetica Neue', sans-serif;
                font-size: 22px;
                vertical-align: middle;
                color: #5f6368;
                line-height: 24px;
                padding-left: 4px;
            }
            .Nav-search {
                display: flex;
                height: 48px;
                flex-grow: 1;
                flex-shrink: 1;
                flex-basis: 100%;
            }
            .Nav-search-form {
                background: #f1f3f4;
                border: 1px solid transparent;
                -webkit-border-radius: 8px;
                border-radius: 8px;
                display: flex;
                flex: 1 1 auto;
                margin-right: auto;
                max-width: 720px;
                padding-left: 10px;
                padding-right: 30px;
                position: relative;
                -webkit-transition: background 100ms ease-in,width 100ms ease-out;
                transition: background 100ms ease-in,width 100ms ease-out;
            }
            .Nav-search-form input {
                background-color: #f1f3f4;
                border: none;
                font-size: 1rem;
                margin-left: 46px;
            }
            .Nav-user {
                height: 48px;
                flex-grow: 0;
                flex-shrink: 0;
                flex-basis: auto;
                display: flex;
                justify-content: center;
                flex-direction: column;
            }
            .Nav ul {
                display: flex;
                justify-content: space-between;
            }
            .Nav &gt; ul {
                padding: 4px 16px;
            }
            .Nav li {
                display: flex;
                padding: 6px 8px;
            }
            .Nav a {
                color: #067df7;
                text-decoration: none;
                font-size: 13px;
            }
        
      </style>
    </nav>
  </header>
  <main
    className="App-main"
  >
    <div
      className="App-main-content"
    >
      <div
        className="Spinner"
      >
        <svg
          className="Spinner-spinner"
          height="32px"
          viewBox="0 0 66 66"
          width="32px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            className="Spinner-path"
            cx="33"
            cy="33"
            fill="none"
            r="30"
            strokeLinecap="round"
            strokeWidth="6"
          />
        </svg>
        <style>
          
            .Spinner {
                width: 100%;
                height: 100%;
                display: flex;
                justify-content: center;
                padding-top: 2rem;
            }
            .Spinner-spinner {
            animation: rotator 1.4s linear infinite;
            }
            @keyframes rotator {
            0% {
                transform: rotate(0deg);
            }
            100% {
                transform: rotate(270deg);
            }
            }
            .Spinner-path {
            stroke-dasharray: 187;
            stroke-dashoffset: 0;
            transform-origin: center;
            animation: dash 1.4s ease-in-out infinite, colors 5.6s ease-in-out infinite;
            }
            @keyframes colors {
            0% {
                stroke: #4285f4;
            }
            25% {
                stroke: #de3e35;
            }
            50% {
                stroke: #f7c223;
            }
            75% {
                stroke: #1b9a59;
            }
            100% {
                stroke: #4285f4;
            }
            }
            @keyframes dash {
            0% {
                stroke-dashoffset: 187;
            }
            50% {
                stroke-dashoffset: 46.75;
                transform: rotate(135deg);
            }
            100% {
                stroke-dashoffset: 187;
                transform: rotate(450deg);
            }
            }                
        
        </style>
      </div>
    </div>
  </main>
</div>
`;
